packages/scaffold-ui/src/partials/w3m-connector-list/index.ts

```diff
@@ -303,6 +306,12 @@ export class W3mConnectorList extends LitElement {
 
   private onClickConnector(item: ConnectorItem) {
     const redirectView = RouterController.state.data?.redirectView
+
+    const isFreeTier = this.plan.tier === 'starter' || this.plan.tier === 'none'
+    const hasExceededLimit = this.plan.limits.isAboveRpcLimit || this.plan.limits.isAboveMauLimit
```

Comment:
isn't this repeated every time you check? why don't you move this to a utility method on the controller and call it from all the views that require it?